invite_service/src/invite_service_impl.py

```diff
@@ -0,0 +1,144 @@
+from src.models.invite import Invite
+
+
+class InviteServiceImpl(GrpcServicer):
+    """
+    Implementation of the Event Service.
+
+    Attributes
+    ----------
+    invites : List[Invite]
+        List of invites
+
+    Methods
+    -------
+    get_invites_by_user_id()
+        Function that need to be bind to the server that returns invites list.
+    create_invite()
+        Function that need to be bind to the server that creates the invite.
+    update_invite()
+        Function that need to be bind to the server that updates the invite.
+    delete_invite()
+        Function that need to be bind to the server that deletes the invite.
+    """
+
+    def __init__(self):
+        self.invites = [
+            Invite(
+                "id", "event_id", "author_id", "invitee_id", proto.InviteStatus.PENDING
+            ),
+            Invite(
+                "id", "event_id", "author_id", "invitee_id", proto.InviteStatus.PENDING
+            ),
+        ]
+
+    def get_invites_by_user_id(
+        self, request: proto.InvitesByUserIdRequest, context: grpc.ServicerContext
+    ) -> proto.InvitesResponse:
+        """
+        Get all invites by user id.
+
+        Parameters
+        ----------
+        request : InvitesByUserIdRequest
+            Request data.
+        context : grpc.ServicerContext
+            Request context.
+
+        Returns
+        -------
+        InvitesResponse
+            Invites object for invite response.
+        """
+        return proto.InvitesResponse(
+            code=200,
+            invites=proto.ListOfInvites(
+                invites=[invite.to_grpc_invite() for invite in self.invites]
+            ),
+        )
+
+    def get_invites_by_invitee_id(
+        self, request: proto.GetInviteeByInviteIdRequest, context: grpc.ServicerContext
+    ) -> proto.InvitesResponse:
+        """
+        Get all invites by invitee id.
+
+        Parameters
+        ----------
+        request : InviteeByInviteIdRequest
+            Request data.
+        context : grpc.ServicerContext
+            Request context.
+
+        Returns
+        -------
+        InvitesResponse
+            Invites object for invite response.
+        """
+        pass
+
+    def create_invite(
+        self, request: proto.Invite, context: grpc.ServicerContext
+    ) -> proto.BaseResponse:
+        """
+        Create invite.
+
+        Parameters
+        ----------
+        request : GrpcInvite
+            Request data containing GrpcInvite.
+        context : grpc.ServicerContext
+            Request context.
+
+        Returns
+        -------
+        BaseResponse
+            Object containing status code and message if the response status is not 200.
+        """
+        pass
+
+    def update_invite(
+        self, request: proto.Invite, context: grpc.ServicerContext
+    ) -> proto.BaseResponse:
+        """
+        Update invite.
+
+        Parameters
+        ----------
+        request : GrpcInvite
```

Comment:
Args types in annotations and docstring are different. Remove "GrpcInvite", add "proto.Invite" instead in docstring to fix